src/index.ts

```diff
@@ -142,9 +146,23 @@ export function withSpid(
     samlConfig
   );
 
-  return loadSpidStrategyOptions()
+  return fromNullable(null)(appConfig.startupIdpsMetadata)
+    .map(parseStartupSpidStrategy)
+    .fold(
+      () => loadSpidStrategyOptions(),
+      idpOptionsRecord =>
+        new Task(() =>
+          Promise.resolve(
+            bindSpidStrategyOptions(
```

Comment:
naming: usually "bind" prefix is used to bind _methods_ to some data. here we're just making another object. Probably "makeSpidStrategyOptions" suits better here.

Reply:
Fixed in a4bdb13